src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -843,11 +843,14 @@ open_archive(const xrt_core::device* device)
   
   try {
     std::string archive_path = xrt_core::device_query<xrt_core::query::archive_path>(device);
-    auto full_archive_path = xrt_core::environment::platform_path(archive_path).string();
+    if (archive_path.empty()) {
+      // If shim does not provide an archive path (e.g. alveo shim), ignore the error
+      return nullptr;
+    }
+    std::string full_archive_path = xrt_core::environment::platform_path(archive_path).string();
     archive = std::make_unique<xrt_core::archive>(full_archive_path);
-  } catch (const std::exception& /*e*/) {
-    // Continue without archive - this is not a fatal error
+  } catch (const std::exception& e) {
+    std::cerr << "Error opening archive: " << e.what() << std::endl;
```

Comment:
We need to expand on this error and direct the user what to do.  The code (platform_path) already is checks a few locations, one being custom to the current XRT build 

- `$HOME/.local/share/xrt/<major>.<minor>.<patch>/amdxdna/bins`

I suggest, but I am not sure what is best,  that we make the catch Linux specific and direct the user to run an install script from VTD to copy the proper archive to to this version specific location.  Let discuss.

Reply:
Yes.  Let's do what you suggested.  Can you add a script under core/tools/xbutil2 that can install the proper archive under `$HOME/.local/share/xrt/<major>.<minor>.<patch>/amdxdna/bins`?   I will sort out where we install this script and then we can amend the message in the catch statement accordingly.  I think you need some code in XBUtilities to compute the proper arguments to the script, like XRT `major.minor.patch` from a compile time define from CMake and the base archive name?